### PR TITLE
PHOENIX-6641 Bump Guava to 31.0.1 in phoenix-thirdparty

### PR DESCRIPTION
**Pull request by stoty**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -68,7 +68,7 @@
 
   <properties>
     <rename.offset>org.apache.phoenix.thirdparty</rename.offset>
-    <guava.version>29.0-android</guava.version>
+    <guava.version>31.0.1-android</guava.version>
```
Comment (ankitsinghal) [inline pom.xml:71]: Do we still want to stick to android not 31.0.1-jre because of java 7 compatibility? Not sure if we add any compat module in our third-party library for jdk7 separately though this is entirely unrelated to this fix.

**Review by ankitsinghal — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (stoty) [reply pom.xml:71]: Exactly.
There is no special provision in thridparty for Java 7, apart from using guava-android.
